### PR TITLE
suspend test suite refactor

### PR DESCRIPTION
**Pull request by sulewskiprzemyslaw**
Signed-off-by: Przemyslaw Sulewski <przemyslaw.sulewski@3mdeb.com>

### CODE REVIEW

**Review by Pokisiekk — APPROVED** [review]

File: docs/unified-test-documentation/dasharo-performance/407-cyclic-suspend-and-resume.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Dasharo Performance: Cyclic platform suspend and resume
```
Comment (miczyg1) [inline docs/unified-test-documentation/dasharo-performance/407-cyclic-suspend-and-resume.md:1]: Cyclic suspend resume test is made for reliability not performance reasons. It fits into the compatibility category.

If we want suspend/resume the performance test, I would focus on "how much time does it take from issuing a waking vector to see OS login prompt or screensaver back again"

tl;dr

Please move the file content to the 31M file in compatibility as a new test case number

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-performance/407-cyclic-suspend-and-resume.md (hunk 1 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-performance/407-cyclic-suspend-and-resume.md:1]: @miczyg1 OK, I just moved this test case and remove references to it.

BTW, in my opinion, we should think about the new division of the test groups. At this point, I wanted to assume that the group `performance` contains not only performative tests but also reliability tests. Maybe I should prepare the proposal for the new division (ofc with a description that informs how we allocate a given test to a given group).

File: docs/unified-test-documentation/dasharo-compatibility/31M-platform-suspend-and-resume.md [hunk 2]
```diff
@@ -1,60 +1,506 @@
...
+1. After entering the first command the DUT should enter sleep mode.
+1. The output of the second command should contain information about suspend
+    mode, especially information about moments of entering and leaving it. Also,
+    it should not contain information that any suspend and awake hook fails.
+    The example output from this command has been shown in the
+    [Summary example output](#summary-example-output) section.
+
+## Summary example output
+
+Below is shown an example output from the command `cat /var/log/pm-suspend.log`:
```
[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31M-platform-suspend-and-resume.md]: @miczyg1 I just change the commands for obtaining the results of suspend/resume process and shorten the expected output sections

File: docs/unified-test-documentation/dasharo-performance/407-cyclic-suspend-and-resume.md (hunk 1 above)
Comment (miczyg1) [reply docs/unified-test-documentation/dasharo-performance/407-cyclic-suspend-and-resume.md]: One can do performance tests only when everything works reliably. So first comes reliability, then performance, not the other way around. Performance should check what it really means: speed, time, efficiency, etc.

File: docs/unified-test-documentation/dasharo-compatibility/31M-platform-suspend-and-resume.md [hunk 3]
```diff
@@ -1,60 +1,506 @@
...
+/usr/lib/pm-utils/sleep.d/00logging resume suspend: success.
+
+Running hook /usr/lib/pm-utils/sleep.d/000record-status resume suspend:
+/usr/lib/pm-utils/sleep.d/000record-status resume suspend: success.
+
+Running hook /usr/lib/pm-utils/sleep.d/000kernel-change resume suspend:
+/usr/lib/pm-utils/sleep.d/000kernel-change resume suspend: success.
+
+wto, 10 sty 2023, 14:32:26 CET: Finished.
+```
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply docs/unified-test-documentation/dasharo-compatibility/31M-platform-suspend-and-resume.md:494]: @sulewskiprzemyslaw do you plan to add the windows test cases now or later?

[the remark this answers is not in the document]
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/31M-platform-suspend-and-resume.md:494]: @miczyg1 later probably. I need time to develop the procedures, which will be easy to use and/or automate

**Review by miczyg1 — APPROVED** [review]
